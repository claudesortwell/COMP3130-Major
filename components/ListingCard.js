import React, { useState } from "react";
import { View, Image, Text, TouchableHighlight } from "react-native";
import Shadows from "../styles/shadow";
import TextStyle from "../styles/text";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import { AntDesign } from "@expo/vector-icons";
import { Entypo } from "@expo/vector-icons";
import { MaterialIcons } from "@expo/vector-icons";
import { LinearGradient } from "expo-linear-gradient";
import Colors from "../styles/colors";

const ListingCard = ({ data, navigation }) => {
  return (
    <View
      style={{
        backgroundColor: "#fff",
        width: "100%",
        height: 175,
        borderRadius: 5,
        ...Shadows.main,
        margin: 10,
        flex: 1,
        flexDirection: "row"
      }}
    >
      <Image
        style={{ width: "50%", height: 175, borderTopLeftRadius: 5, borderBottomLeftRadius: 5 }}
        source={data.image.toString().startsWith("file:/") ? { uri: data.image } : data.image}
      />
      <View style={{ padding: 10, flex: 1 }}>
        <Text numberOfLines={1} style={{ ...TextStyle.H3 }}>
          {data.name}
        </Text>

        <View style={{ flexDirection: "row", marginTop: 8 }}>
          <MaterialCommunityIcons name="map-marker" size={18} />
          <Text numberOfLines={2} style={{ ...TextStyle.H6, paddingLeft: 7, marginBottom: 7, width: "80%" }}>
            {data.address}
          </Text>
        </View>

        <View style={{ flexDirection: "row", marginTop: data.locality ? 2 : 16 }}>
          <AntDesign name="tags" size={18} />
          <Text
            numberOfLines={1}
            style={{ ...TextStyle.H6, paddingLeft: 7, paddingTop: 3, marginBottom: 7, width: "85%" }}
          >
            {data.category}
          </Text>
        </View>

        {data.locality && (
          <View style={{ flexDirection: "row", marginTop: 10 }}>
            <Entypo name="globe" size={14} style={{ paddingLeft: 2 }} />
            <Text numberOfLines={1} style={{ ...TextStyle.H6, paddingLeft: 9, marginTop: 1.5, width: "85%" }}>
              {data.locality}
            </Text>
          </View>
        )}

        <View style={{ flexDirection: "row", marginTop: data.locality ? 10 : 16 }}>
          <MaterialIcons name="star" size={18} color="black" />
          <Text
            numberOfLines={1}
            style={{ ...TextStyle.H6, paddingLeft: 7, paddingTop: 3, marginBottom: 7, width: "85%" }}
          >
            {data.stars}/5
          </Text>
        </View>

        <View style={{ flexDirection: "row" }}>
          <TouchableHighlight
            style={{ borderRadius: 5, flexDirection: "row", marginTop: 3 }}
            onPress={() => {
              if (navigation) navigation.navigate("Listing", { id: data.id });
            }}
          >
            <LinearGradient
              colors={[Colors.teal, Colors.blue]}
              style={{ ...Shadows.main, borderRadius: 5, width: "100%" }}
              start={{ x: 0, y: 0 }}
              end={{ x: 1, y: 1 }}
            >
              <Text style={{ ...TextStyle.H5, color: Colors.white, padding: 5, textAlign: "center" }}>View More</Text>
            </LinearGradient>
          </TouchableHighlight>
        </View>
      </View>
    </View>
  );
};

export default ListingCard;
