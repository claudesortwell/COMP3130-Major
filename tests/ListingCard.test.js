import React from "react";
import renderer from "react-test-renderer";
import ListingCard from "../components/ListingCard";

// test("Listing Card Renders", () => {
//   const json = renderer.create(<ListingCard />).toJson();
//   console.log(json);
// });

jest.mock("@expo/vector-icons");

describe("<Listing Card/>", () => {
  const card = renderer.create(
    <ListingCard
      data={{
        name: "McDonald's Sydney",
        address: "135 John Street",
        image: require("../assets/listing1.jpg"),
        category: "Hotel",
        locality: "American",
        stars: 4
      }}
      navigation={{}}
    ></ListingCard>
  );

  const cardDifferent = renderer.create(
    <ListingCard
      data={{
        name: "McDonald's Apple Sydney",
        address: "136 John Street",
        image: require("../assets/listing2.jpg"),
        category: "Tourism",
        locality: "Korean",
        stars: 2.5
      }}
      navigation={{}}
    ></ListingCard>
  );

  const cardEmptyFragment = renderer.create(<ListingCard data={{ name: "" }} navigation={{}}></ListingCard>);

  it("Renders correctly", () => {
    expect(card).toMatchSnapshot();
  });

  it("Renders different data correctly", () => {
    expect(cardDifferent).toMatchSnapshot();
  });

  it("Renders empty fragment if missing data", () => {
    expect(cardEmptyFragment).toMatchSnapshot();
  });
});
