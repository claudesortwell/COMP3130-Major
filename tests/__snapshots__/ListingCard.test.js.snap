// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Listing Card/> Renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 5,
      "elevation": 6,
      "flex": 1,
      "flexDirection": "row",
      "height": 175,
      "margin": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 4.65,
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "borderBottomLeftRadius": 5,
        "borderTopLeftRadius": 5,
        "height": 175,
        "width": "50%",
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "padding": 10,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#252F3F",
          "fontFamily": "Inter_700Bold",
          "fontSize": 14,
        }
      }
    >
      McDonald's Sydney
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 8,
        }
      }
    >
      <Text
        numberOfLines={2}
        style={
          Object {
            "color": "#252F3F",
            "fontFamily": "Inter_500Medium",
            "fontSize": 10,
            "marginBottom": 7,
            "paddingLeft": 7,
            "width": "80%",
          }
        }
      >
        135 John Street
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 2,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#252F3F",
            "fontFamily": "Inter_500Medium",
            "fontSize": 10,
            "marginBottom": 7,
            "paddingLeft": 7,
            "paddingTop": 3,
            "width": "85%",
          }
        }
      >
        Hotel
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 10,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#252F3F",
            "fontFamily": "Inter_500Medium",
            "fontSize": 10,
            "marginTop": 1.5,
            "paddingLeft": 9,
            "width": "85%",
          }
        }
      >
        American
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 10,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#252F3F",
            "fontFamily": "Inter_500Medium",
            "fontSize": 10,
            "marginBottom": 7,
            "paddingLeft": 7,
            "paddingTop": 3,
            "width": "85%",
          }
        }
      >
        4
        /5
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderRadius": 5,
            "flexDirection": "row",
            "marginTop": 3,
          }
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                4279680458,
                4282352632,
              ],
              "endPoint": Array [
                1,
                1,
              ],
              "locations": undefined,
              "startPoint": Array [
                0,
                0,
              ],
            }
          }
          style={
            Object {
              "borderRadius": 5,
              "elevation": 6,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 4.65,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontFamily": "Inter_500Medium",
                "fontSize": 12,
                "padding": 5,
                "textAlign": "center",
              }
            }
          >
            View More
          </Text>
        </ViewManagerAdapter_ExpoLinearGradient>
      </View>
    </View>
  </View>
</View>
`;

exports[`<Listing Card/> Renders different data correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 5,
      "elevation": 6,
      "flex": 1,
      "flexDirection": "row",
      "height": 175,
      "margin": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 4.65,
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "borderBottomLeftRadius": 5,
        "borderTopLeftRadius": 5,
        "height": 175,
        "width": "50%",
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "padding": 10,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#252F3F",
          "fontFamily": "Inter_700Bold",
          "fontSize": 14,
        }
      }
    >
      McDonald's Apple Sydney
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 8,
        }
      }
    >
      <Text
        numberOfLines={2}
        style={
          Object {
            "color": "#252F3F",
            "fontFamily": "Inter_500Medium",
            "fontSize": 10,
            "marginBottom": 7,
            "paddingLeft": 7,
            "width": "80%",
          }
        }
      >
        136 John Street
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 2,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#252F3F",
            "fontFamily": "Inter_500Medium",
            "fontSize": 10,
            "marginBottom": 7,
            "paddingLeft": 7,
            "paddingTop": 3,
            "width": "85%",
          }
        }
      >
        Tourism
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 10,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#252F3F",
            "fontFamily": "Inter_500Medium",
            "fontSize": 10,
            "marginTop": 1.5,
            "paddingLeft": 9,
            "width": "85%",
          }
        }
      >
        Korean
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 10,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#252F3F",
            "fontFamily": "Inter_500Medium",
            "fontSize": 10,
            "marginBottom": 7,
            "paddingLeft": 7,
            "paddingTop": 3,
            "width": "85%",
          }
        }
      >
        2.5
        /5
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderRadius": 5,
            "flexDirection": "row",
            "marginTop": 3,
          }
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                4279680458,
                4282352632,
              ],
              "endPoint": Array [
                1,
                1,
              ],
              "locations": undefined,
              "startPoint": Array [
                0,
                0,
              ],
            }
          }
          style={
            Object {
              "borderRadius": 5,
              "elevation": 6,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 4.65,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontFamily": "Inter_500Medium",
                "fontSize": 12,
                "padding": 5,
                "textAlign": "center",
              }
            }
          >
            View More
          </Text>
        </ViewManagerAdapter_ExpoLinearGradient>
      </View>
    </View>
  </View>
</View>
`;

exports[`<Listing Card/> Renders empty fragment if missing data 1`] = `null`;
